src/zkevm_specs/evm/execution/blockhash.py

```diff
@@ -0,0 +1,31 @@
+from ..instruction import Instruction, Transition
+from ..table import BlockContextFieldTag
+from ...util import FQ, N_BYTES_U64
+
+def blockhash(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    block_number = instruction.rlc_to_fq(instruction.stack_pop(), N_BYTES_U64)
+    
+    current_block_number = instruction.block_context_lookup(BlockContextFieldTag.Number)
+    is_current_bigger = instruction.compare(block_number, current_block_number.expr(), N_BYTES_U64)[0]
+
+    diff = current_block_number.expr() - block_number if is_current_bigger == 1 else block_number - current_block_number.expr()
+    is_invalid_range = 1 - (
+        is_current_bigger
+        * instruction.compare(diff, FQ(257), N_BYTES_U64)[0]
+    )
+    op = BlockContextFieldTag.HistoryHash
+    block_hash = FQ(0) if is_invalid_range == 1 else instruction.block_context_lookup(op, block_number)
```

Comment:
You probably can't use the `Range256` variant in `FixedTable` to check whether or not the difference between `current_block_number` and `block_number` is `< 256`. Mainly because you want to handle the case where the lookup fails, i.e. simply push `0` to the stack.

So I think you will need a comparison gadget to check that `block_number < current_block_number`.

So you would have something like:
```python
# get value that was pushed to stack (RLC-encoded)
block_hash = instruction.stack_push()

# comparing block_number and current_block_number
block_lt, _ = instruction.compare(block_number, current_block_number, 8)
diff_lt, _ = instruction.compare(current_block_number - block_number, FQ(256), 2)

# get the expected block hash depending on the above conditions (RLC-encoded)
expected_block_hash = instruction.select(
    block_lt * diff_lt,
    instruction.block_context_lookup(
        BlockContextFieldTag.HistoryHash,
        block_number,
    ),
    FQ.zero(),
)

# block hash is as expected
instruction.constrain_equal(block_hash, expected_block_hash)
```

Reply:
That's exactly what I want to know. Thank you for your comment!

Reply:
@roynalnaruto There is an error from `block_conext_lookup()` in `select` instruction. Because `instruction.select()` function is called after calculating argument, so `block_context_lookup` with invalid `block_number` raise `LookupUnsatFailure`. Do you have any idea for handling this situation? Thank you.

Reply:
Ah, of course. Yes that's expected.

In that case, you could scrap the `instruction.select(..)` and simply have:
```python
if block_lt * diff_lt == FQ.one():
    expected_block_hash = instruction.block_context_lookup(
        BlockContextFieldTag.HistoryHash,
        block_number,
    ),
else:
    expected_block_hash = FQ.zero()
```

Reply:
Thanks a lot 👍🏼

Reply:
@jeong0982 you will need to fix the formatting:

* To check which file needs formatting
```
$ make lint
```
* Formatting
```
$ black src/
$ black tests/
```

Reply:
Ok. Thank you!